app/utils/code-mirror-themes.ts

```diff
@@ -1,51 +1,246 @@
-  },
-  '.cm-foldGutter .cm-gutterElement': {
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-    fontSize: '0.875em',
-    minWidth: '1.2rem',
-    color: '#94a3b8',
+
+  '.cm-lineNumbers': {
+    '& .cm-gutterElement': {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'flex-end',
+      padding: '0 0.5rem 0 1rem',
+      fontSize: '0.875em',
+      color: '#94a3b8',
+    },
+  },
+
+  '.cm-foldGutter': {
+    '& .cm-gutterElement': {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      fontSize: '0.875em',
+      minWidth: '1.2rem',
+      color: '#94a3b8',
+    },
+  },
+
+  '.cm-content': {
+    padding: '0.5rem 0',
   },
+
+  '.cm-collapsedLines': {
+    height: '1.75rem', // h7
+    paddingLeft: '74px',
+    borderTopWidth: '1px',
+    borderBottomWidth: '1px',
+    fontSize: '0.75rem', // text-xs
+    fontFamily: 'Montserrat, sans-serif',
+    lineHeight: '1rem', // text-xs
+    background: tailwindColors.sky['50'],
+    color: tailwindColors.sky['700'],
+    borderColor: tailwindColors.sky['100'],
+
+    '&:before': {
+      content: '"Expand"',
+      position: 'absolute',
+      height: '1.75rem',
+      paddingLeft: '326px',
+      marginLeft: '-368px',
+      fontSize: '0.75rem',
+      lineHeight: '1rem',
+      backgroundColor: tailwindColors.sky['50'],
+      backgroundImage: 'url("/assets/images/codemirror/expand-diff-middle.svg")',
+      backgroundRepeat: 'no-repeat',
+      backgroundPosition: '306px center',
+      borderColor: tailwindColors.sky['100'],
+      borderTopWidth: '1px',
+      borderBottomWidth: '1px',
+      paddingTop: '5px',
+      paddingBottom: '5px',
+      marginTop: '-6px',
+      backgroundSize: '12px 12px',
+      zIndex: '200',
+    },
+
+    '&:after': {
+      content: '" "',
+      position: 'absolute',
+      marginLeft: '-5px',
+      width: '20px',
+      fontSize: '0.75rem',
+      lineHeight: '1rem',
+      backgroundColor: tailwindColors.sky['50'],
+    },
+
+    '&:hover': {
+      background: tailwindColors.sky['100'],
+      color: tailwindColors.sky['800'],
+
+      '&:before': {
+        backgroundColor: tailwindColors.sky['100'],
+        color: tailwindColors.sky['800'],
+      },
+
+      '&:after': {
+        backgroundColor: tailwindColors.sky['100'],
+        color: tailwindColors.sky['800'],
+      },
+    },
+
+    '&:first-child': {
+      borderTop: 'none',
+      marginTop: '-0.5rem',
+      marginBottom: '0.5rem',
+
+      '&:before': {
+        borderTop: 'none',
+        marginTop: '-5px',
+        backgroundImage: 'url("/assets/images/codemirror/expand-diff-top.svg")',
+      },
+    },
+
+    '&:last-child': {
+      borderBottom: 'none',
+      marginTop: '0.5rem',
+      marginBottom: '-0.5rem',
+
+      '&:before': {
+        borderBottom: 'none',
+        backgroundImage: 'url("/assets/images/codemirror/expand-diff-bottom.svg")',
+      },
+    },
+  },
```

Comment:
What has to be kept in mind is that although this is a TypeScript file, the contents of it are written using https://github.com/marijnh/style-mod, which is a standardized way to describe CSS using JS.

And so, since this file is essentially a stylesheet, it has to be treated as a CSS, and adding too much javascript complexity to it will quickly make it unmanagable.